src/main/java/org/broadinstitute/hellbender/tools/funcotator/Funcotator.java

```diff
@@ -415,4 +272,372 @@ private void enqueueAndHandleVariant(final VariantContext variant, final Referen
+        final Map<Path, Properties> metaData = new LinkedHashMap<>();
+
+        boolean hasGencodeDataSource = false;
+
+        // Go through our directories:
+        final Set<String> names = new HashSet<>();
+        for ( final String pathString : dataSourceDirectories ) {
+            final Path p = IOUtils.getPath(pathString);
+            if ( !isValidDirectory(p) ) {
+                logger.warn("WARNING: Given path is not a valid directory: " + p.toUri().toString());
+                continue;
+            }
+
+            // Now that we have a valid directory, we need to grab a list of sub-directories in it:
+            try {
+                for ( final Path dataSourceTopDir : Files.list(p).filter(Funcotator::isValidDirectory).collect(Collectors.toSet()) ) {
+
+                    // Get the path that corresponds to our reference version:
+                    final Path dataSourceDir = dataSourceTopDir.resolve(refVersion.toString());
+
+                    // Make sure that we have a good data source directory:
+                    if ( isValidDirectory(dataSourceDir) ) {
+
+                        // Get the config file path:
+                        final Path configFilePath = getConfigfile(dataSourceDir);
+
+                        // Read the config file into Properties:
+                        final Properties configFileProperties = readConfigFileProperties(configFilePath);
+
+                        // Validate the config file properties:
+                        assertConfigFilePropertiesAreValid(configFileProperties, configFilePath);
+
+                        // Make sure we only have 1 of this data source:
+                        if ( names.contains(configFileProperties.getProperty("name")) ) {
+                            throw new UserException.BadInput("Error: contains more than one dataset of name: "
+                                    + configFileProperties.getProperty("name") + " - one is: " + configFilePath.toUri().toString());
+                        }
+                        else {
+                            names.add( configFileProperties.getProperty("name") );
+
+                            if ( FuncotatorArgumentDefinitions.DataSourceType.getEnum(configFileProperties.getProperty("type")) ==
+                                    FuncotatorArgumentDefinitions.DataSourceType.GENCODE ) {
+                                hasGencodeDataSource = true;
+                            }
+
+                            // Add our config file to the properties list:
+                            metaData.put( configFilePath, configFileProperties );
+                        }
+                    }
+                }
+            }
+            catch (final IOException ex) {
+                throw new GATKException("Unable to read contents of: " + p.toUri().toString());
+            }
+        }
+
+        if ( !hasGencodeDataSource ) {
```

Comment:
Does this mean that a GENCODE datasource is required for any run of funcotator?

Reply:
Yes - right now GENCODE is required to make sure that the annotations can be correctly made by genome location.